lib/devise-security/models/compatibility/active_record_patch.rb

```diff
@@ -7,7 +7,11 @@ class NotPersistedError < ActiveRecord::ActiveRecordError; end
 
       module ActiveRecordPatch
         extend ActiveSupport::Concern
-        unless Devise.activerecord51?
+
+        active_record_51 = (Devise.respond_to?(:active_record_51?) && Devise.active_record_51?) ||
```

Comment:
This check seems not to work:

(byebug) ActiveRecord.gem_version
#<Gem::Version "6.1.7.3">

### devise 4.8
> Devise.activerecord51?
true

### devise 4.9.1
> active_record_51 = (Devise.respond_to?(:active_record_51?) && Devise.active_record_51?) || (Devise.respond_to?(:activerecord51?) && Devise.activerecord51?)
false

The whole `unless Devise.activerecord51?` should be removed.
1. `saved_change_to_encrypted_password` seems not to be used
2. `encrypted_password_before_last_save` seems not to be used
3. `will_save_change_to_encrypted_password?` is used for Rails <= 5.0 which is out of maintanance for a long time.
4. If someone uses Rails 5.0 or older `devise-security` can be pinned to a working version.
5. `Devise.activerecord51` is not a public method.

Reply:
Re "The whole unless Devise.activerecord51? should be removed.", #415 does this.

Reply:
👍  Thanks. I've updated the PR to use the underlying implementation of `Device.active_record_51?`. I can also remove the entire patch if it's indeed unnecessary, but I'm not familiar enough with the codebase to tell for sure if that's the case.

cc @dillonwelch @olbrich if you have any recommendations

Reply:
- Dropping old rails version support is outside the scope of this PR
- I think this is the only place where this method was used or needed, so this should be fine.